### PR TITLE
Issue #31002 Enable RPC for all-clusters-app

### LINKED ISSUE #31002: [Tooling] Enable RPC for all-clusters-app
### Reproduction steps / Feature

RPC is not available for the all-clusters-app. Most of the existing protos are relevant for the all-clusters-app. Would be nice if we could build the app with RPC:

Is there any specific reason why it is not added?

### Platform

other

### Platform Version(s)

master

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by adis-ikea**
Fixes #31002 Adding RPC support for all-clusters-app

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by tehampson — APPROVED** [review]

File: examples/all-clusters-app/linux/BUILD.gn [hunk 1]
```diff
@@ -57,7 +64,52 @@ source_set("chip-all-clusters-common") {
...
+      "PW_RPC_TRACING_SERVICE=1",
+    ]
+
+    sources += [
+      "${chip_root}/examples/platform/linux/Rpc.cpp",
+      "${chip_root}/examples/platform/linux/system_rpc_server.cc",
+    ]
+
+    deps += [
+      "$dir_pw_hdlc:pw_rpc",
```
Comment (mspang) [inline examples/all-clusters-app/linux/BUILD.gn:84]: This is begging for better abstraction (ie, define a common library or at the very least a reusable template())

Comment (adis-ikea) [reply examples/all-clusters-app/linux/BUILD.gn]: I moved most of the rpc stuff to a new file. Is this OK for you or you had something else in mind?

Comment (adis-ikea) [reply examples/all-clusters-app/linux/BUILD.gn]: @mspang any feedback?

Comment (andy31415) [reply examples/all-clusters-app/linux/BUILD.gn]: Closing no-feedback since Dec

**Pull request merged**